x/delayedack/ibc_middleware.go

```diff
@@ -116,13 +117,15 @@ func (im IBCMiddleware) OnAcknowledgementPacket(
 	}
 	logger := ctx.Logger().With("module", "DelayedAckMiddleware")
 
+	rollappPortOnHub, rollappChannelOnHub := packet.SourcePort, packet.SourceChannel
+
 	var ack channeltypes.Acknowledgement
 	if err := types.ModuleCdc.UnmarshalJSON(acknowledgement, &ack); err != nil {
 		logger.Error("Unmarshal acknowledgement", "err", err)
 		return errorsmod.Wrapf(sdkerrors.ErrUnknownRequest, "unmarshal ICS-20 transfer packet acknowledgement: %v", err)
 	}
 
-	rollappID, transferPacketData, err := im.ExtractRollappIDAndTransferPacket(ctx, packet)
+	rollappID, transferPacketData, err := im.ExtractRollappIDAndTransferPacket(ctx, packet, rollappPortOnHub, rollappChannelOnHub)
```

Comment:
I think `RollappPacket` should have `GetPacketUID` method
than u don't need to change `im.ExtractRollappIDAndTransferPacket(ctx, packet)` interface

Reply:
actually the packet here is `channeltypes.Packet` not rollappPacket.